src/Fantomas.Core/CodePrinter.fs

```diff
@@ -753,33 +753,40 @@ let genExpr (e: Expr) =
         |> genNode node
     | Expr.Dynamic node -> genExpr node.FuncExpr +> !- "?" +> genExpr node.ArgExpr |> genNode node
     | Expr.PrefixApp node ->
-        let fallback = genSingleTextNode node.Operator +> genExpr node.Expr
+        let genWithoutSpace = genSingleTextNode node.Operator +> genExpr node.Expr
+        let genWithSpace = genSingleTextNode node.Operator +> sepSpace +> genExpr node.Expr
 
         match node.Expr with
-        | Expr.Constant _
-        | Expr.InterpolatedStringExpr _ when not (String.startsWithOrdinal "%" node.Operator.Text) ->
-            genSingleTextNode node.Operator +> sepSpace +> genExpr node.Expr
+        // E.g. + <@ 1 @>
+        | Expr.Quote _ -> genWithSpace
+        | Expr.Constant(constant) ->
+            // E.g. !! @"foobar", because !!@ would be mistaken for an operator.
+            match constant with
+            | Constant.FromText(stn) ->
+                match stn.Text.[0] with
+                | '@'
+                | '-'
+                | '+' -> genWithSpace
+                | _ -> genWithoutSpace
+            // E.g. - +1<m>
+            | Constant.Measure _ -> genWithSpace
```

Comment:
There doesn't need to be a space if the measure literal doesn't start with `+` or `-`, though.